apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/sampling/SamplingService.java

```diff
@@ -90,9 +88,9 @@ public void shutdown() {
      * @return true, if sampling mechanism is on, and getDefault the sampling factor successfully.
      */
     public boolean trySampling(String operationName) {
-        if (on) {
+        if (samplingRateWatcher.getSamplingRate() > 0) {
             int factor = samplingFactorHolder.get();
-            if (factor < Config.Agent.SAMPLE_N_PER_3_SECS) {
+            if (factor < samplingRateWatcher.getSamplingRate()) {
```

Comment:
Multiple `get`(s) cause unnecessary resources. Because this get is reading from the memory rather than CPU cache, due to the AtomicInteger.